packages/vitest/src/node/reporters/html.ts

```diff
@@ -0,0 +1,79 @@
+import { existsSync, promises as fs } from 'fs'
+import { dirname, relative, resolve } from 'pathe'
+import fg from 'fast-glob'
+import { stringify } from 'flatted'
+import { distDir } from '../../constants'
+import type { Vitest } from '../../node'
+import type { File, Reporter } from '../../types'
+import { getOutputFile } from '../../utils/config-helpers'
+import { getModuleGraph } from '../../api/setup'
+import type { ResolvedConfig } from './../../types/config'
+import type { ModuleGraphData } from './../../types/general'
+
+interface HTMLReportData {
+  paths: string[]
+  files: File[]
+  config: ResolvedConfig
+  moduleGraph: Record<string, ModuleGraphData>
+}
+
+export class HTMLReporter implements Reporter {
+  start = 0
+  ctx!: Vitest
+
+  onInit(ctx: Vitest): void {
+    this.ctx = ctx
+    this.start = Date.now()
+  }
+
+  async onFinally() {
+    const result: HTMLReportData = {
+      paths: await this.ctx.state.getPaths(),
+      files: this.ctx.state.getFiles(),
+      config: this.ctx.config,
+      moduleGraph: {},
+    }
+    await Promise.all(
+      result.files.map(async (file) => {
+        result.moduleGraph[file.filepath] = await getModuleGraph(this.ctx, file.filepath)
+      }),
+    )
+    await this.writeReport(stringify(result))
+  }
+
+  /**
+   * Writes the report to an output file if specified in the config,
+   * or logs it to the console otherwise.
```

Comment:
I don't think this doc is true. It doesn't write it to the console, and I don't think if it should

Reply:
yes. Copy it from json report. forget to delete the comment. 🤦